### ISSUE TITLE
Unable to push or create package version with static resource containing file where filename starts with `.`

### ISSUE THREAD

**Issue by coolbulevar** [forcedotcom/cli#2666]
### Summary
Files with name starting with `.` ignored by `sf project deploy` and `sf package version create` commands.

It seems the issue is not related to [general issues with node 18.16.0 #2125](https://github.com/forcedotcom/cli/issues/2125) and as suggested by @mshanemc opening new issue.

### Steps To Reproduce
Repository that's configured to reproduce the issue.
[cli-zip-static-issue](https://github.com/coolbulevar/cli-zip-static-issue)

- create default scratch
- push code with `sf project deploy start`

### Expected result
`normalFile.xml` and `.specialFile` are being pushed successfully.

### Actual result
only `normalFile.xml` is pushed and present in static resource.

### System Information
```json
{
  "architecture": "linux-x64",
  "cliVersion": "@salesforce/cli/2.25.7",
  "nodeVersion": "node-v20.9.0",
  "osVersion": "Linux 5.15.0-91-generic",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.5 (core)",
    "@oclif/plugin-commands 3.1.1 (core)",
    "@oclif/plugin-help 6.0.12 (core)",
    "@oclif/plugin-not-found 3.0.9 (core)",
    "@oclif/plugin-plugins 4.1.17 (core)",
    "@oclif/plugin-search 1.0.12 (core)",
    "@oclif/plugin-update 4.1.8 (core)",
    "@oclif/plugin-version 2.0.11 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.9 (core)",
    "@oclif/plugin-which 3.0.15 (core)",
    "@salesforce/analytics 1.3.19 (user)",
    "@salesforce/cli 2.25.7 (core)",
    "apex 3.0.18 (core)",
    "auth 3.3.2 (core)",
    "data 3.0.16 (core)",
    "deploy-retrieve 3.2.0 (core)",
    "dev 1.3.2 (user)",
    "info 3.0.17 (core)",
    "limits 3.1.4 (core)",
    "marketplace 1.0.16 (core)",
    "org 3.3.5 (core)",
    "packaging 2.1.1 (core)",
    "schema 3.1.0 (core)",
    "settings 2.0.17 (core)",
    "sobject 1.1.2 (core)",
    "source 3.1.5 (core)",
    "telemetry 3.1.10 (core)",
    "templates 56.0.10 (core)",
    "trust 3.3.2 (core)",
    "user 3.2.3 (core)"
  ]
}
```
### Additional information
For the `sf package version create` command behavior is similar, special file is gone missing.

![image](https://github.com/forcedotcom/cli/assets/10941673/53e3d9b7-3174-40a3-a97c-dc4eea9460ed)

UPD 
Updated to the latest version of Salesforce CLI and confirm that still seeing issue.

UPD2
As a workaround we are still using `sfdx-cli@7.203.6` on `node-version: 16.20.0`

**Comment by coolbulevar** [comment]
Updated to the latest version of Salesforce CLI and confirm that still seeing issue.

```json
{
  "architecture": "linux-x64",
  "cliVersion": "@salesforce/cli/2.25.7",
  "nodeVersion": "node-v20.9.0",
  "osVersion": "Linux 5.15.0-91-generic",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.5 (core)",
    "@oclif/plugin-commands 3.1.1 (core)",
    "@oclif/plugin-help 6.0.12 (core)",
    "@oclif/plugin-not-found 3.0.9 (core)",
    "@oclif/plugin-plugins 4.1.17 (core)",
    "@oclif/plugin-search 1.0.12 (core)",
    "@oclif/plugin-update 4.1.8 (core)",
    "@oclif/plugin-version 2.0.11 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.9 (core)",
    "@oclif/plugin-which 3.0.15 (core)",
    "@salesforce/analytics 1.3.19 (user)",
    "@salesforce/cli 2.25.7 (core)",
    "apex 3.0.18 (core)",
    "auth 3.3.2 (core)",
    "data 3.0.16 (core)",
    "deploy-retrieve 3.2.0 (core)",
    "dev 1.3.2 (user)",
    "info 3.0.17 (core)",
    "limits 3.1.4 (core)",
    "marketplace 1.0.16 (core)",
    "org 3.3.5 (core)",
    "packaging 2.1.1 (core)",
    "schema 3.1.0 (core)",
    "settings 2.0.17 (core)",
    "sobject 1.1.2 (core)",
    "source 3.1.5 (core)",
    "telemetry 3.1.10 (core)",
    "templates 56.0.10 (core)",
    "trust 3.3.2 (core)",
    "user 3.2.3 (core)"
  ]
}
```

**Comment by mshanemc** [comment]
Nice repro, thanks!

**Comment by mshanemc** [comment]
OK, we have a PR open for this.  We've always had `**/.*` in `.forceignore`'s  defaults, and that's what makes it ignore your file.  If you have a forceignore, we add your file's contents to our defaults.

Therefore, if you want to not have that behavior this PR will allow you to add `!**/.*` to your `.forceignore`.  I'd suggest being more selective than that (in your case, something like `!**/staticresources/**/.*`) just because hidden files end up all over the place.

**Comment by jshackell-sfdc** [comment]
This issue is fixed in version 2.28.6.
